src/engine/Service.cpp

```diff
@@ -94,13 +94,18 @@ size_t Service::getCostEstimate() {
 }
 
 // ____________________________________________________________________________
-ProtoResult Service::computeResult([[maybe_unused]] bool requestLaziness) {
+ProtoResult Service::computeResult(bool requestLaziness) {
   // Try to simplify the Service Query using it's sibling Operation.
   if (auto valuesClause = getSiblingValuesClause(); valuesClause.has_value()) {
-    auto openBracketPos = parsedServiceClause_.graphPatternAsString_.find('{');
-    parsedServiceClause_.graphPatternAsString_ =
-        "{\n" + valuesClause.value() + '\n' +
-        parsedServiceClause_.graphPatternAsString_.substr(openBracketPos + 1);
+    auto& graphPattern = parsedServiceClause_.graphPatternAsString_;
+    std::string_view patternView{graphPattern};
+    patternView.remove_prefix(graphPattern.find('{'));
```

Comment:
1. Add a comment which case the `ctre::` handles.
2. Start writing comments in general.
3.  This code block is now so long that you can put it in a separate function.
4. This code shouldn't modify the `graphPattern` inplace, but return a new string, otherwise the deep copy semantics are somewhat wrong.